torch/distributed/_functional_collectives.py

```diff
@@ -791,17 +792,17 @@ def _register_ops():
 
 
 def all_gather_tensor_inplace(
-    output: torch.Tensor,
-    input: torch.Tensor,
-    group,  # TODO add a type,
+    output_tensor: torch.Tensor,
+    input_tensor: torch.Tensor,
+    group: ProcessGroup = None,
```

Comment:
functional collective does not only accept a ProcessGroup so please don't add type annotation with ProcessGroup

Reply:
Sounds good - thanks for the heads up. I think explicit typing is still helpful but that can be in another PR

Reply:
Oh I didn't mean to not adding types because other field already have types, I think we should add `RANK_TYPES` like other APIs in this file.